application-services/custom/fledge-export/pkg/transforms/conversion.go

```diff
@@ -0,0 +1,61 @@
+package transforms
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/appcontext"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type Conversion struct {
+}
+
+func NewConversion() Conversion {
+	return Conversion{}
+}
+
+type FledgeReading struct {
+	Timestamp string                 `json:"timestamp"`
+	Asset     string                 `json:"asset"`
+	Readings  map[string]interface{} `json:"readings,omitempty"`
+}
+
+func newFledgeReading(stamp int64, asset string) *FledgeReading {
+	tm := time.Unix(0, stamp*int64(time.Millisecond))
+	reading := FledgeReading{Timestamp: tm.String(), Asset: asset}
+	reading.Readings = make(map[string]interface{})
+	return &reading
+}
+
+// TransformToFledge ...
+func (f Conversion) TransformToFledge(edgexcontext *appcontext.Context, params ...interface{}) (continuePipeline bool, stringType interface{}) {
+	if len(params) < 1 {
+		return false, errors.New("No Event Received")
+	}
+
+	edgexcontext.LoggingClient.Debug("Transforming to Fledge format")
+
+	if event, ok := params[0].(models.Event); ok {
+		payload := make([]FledgeReading, 1)
+		fReading := newFledgeReading(event.Created, event.Device)
+		payload[0] = *fReading
```

Comment:
This makes a copy, i.e. dereferencing the pointer so changes made below will not be in the copy. Simply move this below the readings loop,